src/allmydata/web/directory.py

```diff
@@ -943,8 +965,121 @@ def render_get_name(self, ctx, data):
         ctx.tag.attributes['value'] = name
         return ctx.tag
 
-class ManifestResults(MultiFormatPage, ReloadMixin):
-    docFactory = getxmlfile("manifest.xhtml")
+
+class ReloadableMonitorElement(Element, object):
+    """
+    Like 'ReloadMixin', but for twisted.web.template style. This
+    provides renderers for "reload" and "refesh" and a self.monitor
+    attribute
+    """
+    refresh_time = 60  # 1 minute
+
+    def __init__(self, monitor):
+        self.monitor = monitor
+        super(ReloadableMonitorElement, self).__init__()
+
+    @renderer
+    def refresh(self, req, tag):
+        if self.monitor.is_finished():
+            return ""
+        tag.attributes["http-equiv"] = "refresh"
+        tag.attributes["content"] = str(self.refresh_time)
+        return tag
+
+    @renderer
+    def reload(self, req, tag):
+        if self.monitor.is_finished():
+            return ""
+        # url.gethere would break a proxy, so the correct thing to do is
+        # req.path[-1] + queryargs
```

Comment:
I guess this should also probably use `hyperlink` or similar .. and a general question here, because currently the `Request` objects will be Nevow ones (well, actually, Tahoe overrides that with `MyRequest` but ..) "until everything is ported", so do I just wrap everything in `six.text_type()` .. or is a better pattern to use e.g. `u"{}".format(req.prepath[-1]) so that we don't have a bunch of `six` stuff to get rid of later?

Reply:
I'm going to go with the latter ..